contracts/Core/StakeManager.sol

```diff
@@ -246,27 +245,16 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause,
 
     /// @notice Used by anyone whose lock expired or who lost funds, and want to request withdraw
     // Here we have added penalty to avoid repeating front-run unstake/witndraw attack
-    function resetLock(uint32 stakerId) external initialized whenNotPaused {
-        // Lock should be expired if you want to reset
+    function extendLock(uint32 stakerId) external initialized whenNotPaused {
+        // Lock should be expired if you want to extend
         require(locks[msg.sender][stakers[stakerId].tokenAddress].amount != 0, "Existing Lock doesnt exist");
 
-        Structs.Staker storage staker = stakers[stakerId];
-        uint256 lockedAmount = locks[msg.sender][stakers[stakerId].tokenAddress].amount;
-        IStakedToken sToken = IStakedToken(staker.tokenAddress);
-
-        //Giving out the resetLock penalty
-        uint256 penalty = (lockedAmount * parameters.resetLockPenalty()) / 100;
-        lockedAmount = lockedAmount - penalty;
+        Structs.Lock storage lock = locks[msg.sender][stakers[stakerId].tokenAddress];
 
-        //Calculating the amount of sToken to be minted
-        uint256 sAmount = _convertRZRtoSRZR(lockedAmount, staker.stake, sToken.totalSupply());
-
-        //Updating Staker Stake
-        staker.stake = staker.stake + lockedAmount;
-
-        _resetLock(stakerId);
-
-        require(sToken.mint(msg.sender, sAmount, lockedAmount));
+        //Giving out the extendLock penalty
+        uint256 penalty = (lock.amount * parameters.extendLockPenalty()) / 100;
+        lock.amount = lock.amount - penalty;
+        lock.withdrawAfter = parameters.getEpoch();
```

Comment:
Then the staker will have to wait for ```currentEpoch + withdrawReleasePeriod``` to withdraw due to this condition https://github.com/razor-network/contracts/blob/82db070d2390b3fc9a2bb05e7bf1889b5b4282e3/contracts/Core/StakeManager.sol#L199
One thing i can add here is that you can only extend lock if you have passed the release period, a condition which is not present currently

Reply:
> Then the staker will have to wait for currentEpoch + withdrawReleasePeriod to withdraw due to this condition

yep that's what we mean by extending lock, if we keep current epoch
It won't be extend, its would be like you can withdraw with penalty
not sure if extending is needed, is withdraw with penalty expose us to any issue ?
if not 
we don't even need to have extendLock, why create lock with current epoch.

hmm coming to adding condition, we can do so
earlier we hadn't included it as resetLock could be used in case when delegator lost their sRZR.
now if we are locking in RZR only, its not needed 
so we can add that check

Reply:
hmm cool yep it works out 
we don't have to make user wait to withdraw

Reply:
User calls withdraw 
It fails as lock expired 
Then user call extendLock
We deduct penalty 
User have currentEpoch + WithdrawReleasePeriod to call Withdraw

cool 👍 

Lets add that check only then, that you mentioned above